### PR TITLE
feat(manager): create sequencer events populate sequencers list

### LINKED ISSUE #1015: feat(rotation): add sequencers pubkey to sequencer set on creation
dymint needs to listen to `create_sequencer` event and get the pubkey from it

### PR DESCRIPTION
**Pull request by srene**
# PR Standards

## Opening a pull request should be able to meet the following requirements

--

PR naming convention: https://hackmd.io/@nZpxHZ0CT7O5ngTp0TP9mg/HJP_jrm7A

---

Close #1015 

<-- Briefly describe the content of this pull request -->

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: block/sequencers.go [hunk 1]
```diff
@@ -179,3 +180,27 @@ func (m *Manager) UpdateProposer() error {
...
+		return
+	}
+
+	newSequencer, err := m.SLClient.GetSequencerByAddress(eventData.SeqAddr)
+	if err != nil {
+		m.logger.Error("Unable to add new sequencer from event. err:%w", err)
+		return
+	}
+	sequencers := m.State.Sequencers.Sequencers
+	sequencers = append(sequencers, newSequencer)
```
Comment (mtsitrin) [inline block/sequencers.go:204]:
```suggestion
	sequencers := append(m.State.Sequencers.Sequencers, newSequencer)
```

File: settlement/local/local.go [hunk 2]
```diff
@@ -206,6 +206,11 @@ func (c *Client) GetProposer() *types.Sequencer {
 	return types.NewSequencer(tmPubKey, pubKey.Address().String())
 }
 
+// GetSequencerByAddress returns all sequencer information by its address. Not implemented since it will not be used in mock SL
+func (c *Client) GetSequencerByAddress(address string) (types.Sequencer, error) {
+	return types.Sequencer{}, nil
```
Comment (mtsitrin) [inline settlement/local/local.go:211]: better to return error with empty sequencer?

File: settlement/grpc/grpc.go [hunk 3]
```diff
@@ -228,6 +228,11 @@ func (c *Client) GetProposer() *types.Sequencer {
 	return types.NewSequencer(tmPubKey, pubKey.Address().String())
 }
 
+// GetSequencerByAddress returns all sequencer information by its address. Not implemented since it will not be used in grpc SL
+func (c *Client) GetSequencerByAddress(address string) (types.Sequencer, error) {
+	return types.Sequencer{}, nil
```
Comment (mtsitrin) [inline settlement/grpc/grpc.go:233]: better to return error with empty sequencer?

File: block/manager.go [hunk 4]
```diff
@@ -194,6 +194,7 @@ func (m *Manager) Start(ctx context.Context) error {
 	/* ----------------------------- sequencer mode ----------------------------- */
 	// Subscribe to batch events, to update last submitted height in case batch confirmation was lost. This could happen if the sequencer crash/restarted just after submitting a batch to the settlement and by the time we query the last batch, this batch wasn't accepted yet.
 	go uevent.MustSubscribe(ctx, m.Pubsub, "updateSubmittedHeightLoop", settlement.EventQueryNewSettlementBatchAccepted, m.UpdateLastSubmittedHeight, m.logger)
+	go uevent.MustSubscribe(ctx, m.Pubsub, "newBondedSequencer", settlement.EventQueryNewBondedSequencer, m.UpdateSequencerSet, m.logger)
```
Comment (mtsitrin) [inline block/manager.go:197]: what about full node mode? probably required there as well

**Review by mtsitrin — CHANGES_REQUESTED** [review]

File: block/manager.go (hunk 4 above)
Comment (srene) [reply block/manager.go]: true, moved to before to be called by sequencers and full-nodes

File: settlement/grpc/grpc.go (hunk 3 above)
Comment (srene) [reply settlement/grpc/grpc.go]: added panic because it shouldnt be called from grpc sl

File: settlement/local/local.go (hunk 2 above)
Comment (srene) [reply settlement/local/local.go]: added panic because it shouldnt be called from mock sl

**Review by mtsitrin — APPROVED** [review]

**Pull request merged**